lib/round.rb

```diff
@@ -0,0 +1,52 @@
+class Round
+    attr_reader :deck, :turns, :cards, :number_correct
+    
+    def initialize (deck)
+        @deck = deck
+        @cards = cards
+        @turns = []
+        @number_correct = 0
+        @correct_turns = @correct_turns
```

Comment:
We have a case of inception going on in lines 9 and 10 where we are setting the instance variable to be .. itself? I'm making an assumption, but I think what you are probably trying to do it start these values off as `nil` maybe and define them later on? Lines 30, 42, 46?

Well you can always start these instance variables off as nil, `@instance_variable = nil` or you can just leave them out all together and just define them when you need them in the methods down below where they get defined. While defining instance variables in your initialize is great because you can see them all in one location, this is not the only location where you can define them. They can be defined in any method of the class, the only caveat being you won't have that instance variable available until after the method that defines it gets called.